metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/AbstractReferenceTerm.java

```diff
@@ -1,40 +1,40 @@
 package eu.europeana.enrichment.api.internal;
 
-
 import java.net.URL;
 import java.util.Objects;
 
-public abstract class AbstractReferenceTerm implements ReferenceTerm{
+/**
+ * This class is a basic implementation of {@link ReferenceTerm} that leaves the details of the
+ * candidate types unimplemented.
+ */
+public abstract class AbstractReferenceTerm implements ReferenceTerm {
 
   private final URL reference;
 
-  public AbstractReferenceTerm(URL reference){
+  public AbstractReferenceTerm(URL reference) {
     this.reference = reference;
   }
 
   @Override
-  public URL getReference(){
+  public URL getReference() {
     return reference;
   }
 
   @Override
-  public boolean equals(Object other){
-    if(other == this){
+  public final boolean equals(Object otherObject) {
+    if (otherObject == this) {
       return true;
     }
-
-    if(other == null || getClass() != other.getClass()){
+    if (!(otherObject instanceof ReferenceTerm)) {
```

Comment:
Why change this?
It's comparing here the interface type which it shouldn't and it's not symmetrical.
I think sonar will complain for this too